speed.js

```diff
@@ -16,13 +16,14 @@ module.exports = function(RED) {
 		RED.nodes.createNode(this, config);
 		this.on('input', msg => {
 			speed = speedTest({maxTime: 5000});
+				node.status({fill: "red", shape: "ring", text: "testing ..."});
 
 			speed.on('downloadprogress', progress => {
 				node.status({fill: "yellow", shape: "dot", text: progress + " %"});
 			});
 
 			speed.on('data', data => {
-				msg.payload.speedResults = data;
```

Comment:
Why not have the results in the payload? There are a lot of nodes from node-red that expect data to be in the payload. For example, the file node writes all the contents of the payload into a file. If the Payload no longer contains the speed results they will not be written.